diff --git a/main.tf b/main.tf
index a0b53a1..78b06a4 100644
--- a/main.tf
+++ b/main.tf
@@ -22,6 +22,7 @@ resource "aws_redshift_cluster" "default" {
 
   cluster_parameter_group_name        = join("", aws_redshift_parameter_group.default[*].id)
   automated_snapshot_retention_period = var.automated_snapshot_retention_period
+  manual_snapshot_retention_period    = var.manual_snapshot_retention_period
   port                                = var.port
   cluster_version                     = var.engine_version
   number_of_nodes                     = var.number_of_nodes
diff --git a/variables.tf b/variables.tf
index ea9d0f1..22285dd 100644
--- a/variables.tf
+++ b/variables.tf
@@ -213,3 +213,9 @@ variable "availability_zone_relocation_enabled" {
   default     = false
   description = "Whether or not the cluster can be relocated to another availability zone, either automatically by AWS or when requested. Available for use on clusters from the RA3 instance family"
 }
+
+variable "manual_snapshot_retention_period" {
+  type        = number
+  default     = -1
+  description = "The default number of days to retain a manual snapshot. If the value is -1, the snapshot is retained indefinitely. Valid values are -1 and integers between 1 and 3653"
+}
